### DISCUSSION TITLE
Image: how to disable add a image by pasting

### DISCUSSION THREAD

**Discussion by DengSihan** [ueberdosis/tiptap#2164] [General Questions] [answered]
Thanks a lot for what remarkable jobs you've done!

In [Image](https://tiptap.dev/api/nodes/image) extension 

when user is pasting a image from clipboard which copy from third part domain, such as http://a.com/test.jpg, they may cause a [One-pixel attack](https://www.hindawi.com/journals/wcmc/2021/8891204/).

So, my question is how to prevent add a image by pasting, or how to handle the pasting event customly for image extension?

**Comment by cadars** [comment, marked as answer]
Maybe you could extend the Image extension and rewrite `parseHTML` this way?
I'm not sure, it might become a problem if you parse HTML later.

```js
parseHTML() {
  //return [
  //  {
  //    tag: 'img[src]',
  //  },
  //]
},
```

**Reply by DengSihan** [reply]
Thanks a lot for your reply! @cadars
